Userland/Libraries/LibGUI/FileSystemModel.cpp

```diff
@@ -181,10 +180,29 @@ String FileSystemModel::Node::full_path() const
 
 ModelIndex FileSystemModel::index(String path, int column) const
 {
-    LexicalPath lexical_path(move(path));
+    auto maybe_node = node_for_path(move(path));
+    if (maybe_node.has_value()) {
+        return maybe_node.value()->index(column);
+    } else {
+        return {};
+    }
+}
+
+Optional<const FileSystemModel::Node*> FileSystemModel::node_for_path(String path) const
```

Comment:
`Optional<const FileSystemModel::Node*>` is a strange return type.
We usually use raw pointers for this sort of thing and let the `nullptr` value be the "empty" value.
In the `Optional<T*>` case it's unclear how to interpret a non-empty `nullptr` value.

Reply:
If the optional has a value, `Node*` will never be `nullptr`. I would've preferred passing a `Node const&` instead but that's not allowed by our `Optional` at the moment. If you want, I could make it a `Node const*` and just use `nullptr` instead of optional.